src/main/java/dev/sigstore/fulcio/client/FulcioValidator.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2022 The Sigstore Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package dev.sigstore.fulcio.client;
+
+import com.google.api.client.util.PemReader;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.PublicKey;
+import java.security.cert.*;
+import java.security.spec.EncodedKeySpec;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.X509EncodedKeySpec;
+import java.util.Collections;
+import java.util.Date;
+import org.conscrypt.ct.CTLogInfo;
+import org.conscrypt.ct.CertificateEntry;
+import org.conscrypt.ct.VerifiedSCT;
+
+public class FulcioValidator {
+  private final CTLogInfo ctLogInfo;
+  private final TrustAnchor fulcioRoot;
+
+  public static FulcioValidator newFulcioValidator(byte[] ctfePublicKey, byte[] fulcioRoot)
+      throws InvalidKeySpecException, NoSuchAlgorithmException, CertificateException, IOException,
+          InvalidAlgorithmParameterException {
+    KeyFactory keyFactory = KeyFactory.getInstance("EC");
+    PemReader pemReader =
+        new PemReader(
+            new InputStreamReader(new ByteArrayInputStream(ctfePublicKey), StandardCharsets.UTF_8));
+    PemReader.Section section = pemReader.readNextSection();
+    if (pemReader.readNextSection() != null) {
+      throw new InvalidKeySpecException(
+          "ctfe public key must be only a single PEM encoded public key");
+    }
+    EncodedKeySpec publicKeySpec = new X509EncodedKeySpec(section.getBase64DecodedBytes());
+    PublicKey ctfePublicKeyObj = keyFactory.generatePublic(publicKeySpec);
+
+    CertificateFactory certificateFactory = CertificateFactory.getInstance("X.509");
+    X509Certificate fulcioRootObj =
+        (X509Certificate)
+            certificateFactory.generateCertificate(new ByteArrayInputStream(fulcioRoot));
+
+    TrustAnchor fulcioRootTrustAnchor = new TrustAnchor(fulcioRootObj, null);
+    // this should throw the InvalidAlgorithmException a bit earlier
+    new PKIXParameters(Collections.singleton(fulcioRootTrustAnchor));
+
+    // TODO: this ctloginfo probably needs to be appropriately filled out
+    CTLogInfo ctLogInfo =
+        new CTLogInfo(ctfePublicKeyObj, "fulcio ct log", "intentionally-nonsense");
```

Comment:
random strings?

Reply:
There's a todo sitting there, because I think this needs to be configurable based on the CTLog that fulcio is configured to be using (staging, prod, user-custom)

Reply:
For the third parameter, we could provide a valid URL to the CT log, but I'm hesitant to do so - It's a sharded log, so at some point we'd have to upgrade the java fulcio client to point to the new log. The URL is probably used to confirm the presence of an entry for non-SCT verification. Maybe change `intentionally-nonsense` to `unused-url`?